### PR TITLE
GUIInfo new VideoPlayer.VideoVersionName infolabel

### PR DESCRIPTION
**Pull request by scott967**
Returns the Video Version Name for use in skins

## Description

Adds new infolabel for VideoPlayer.VideoVersionName

## Motivation and context

Provides infolabel to skins to display the version name for a playing movie, for example on seekbar or fullscreeninfo

## How has this been tested?

Compiled in VS for Windows x64 and run with my skin with a label control for $INFO[VideoPlayer.VideoVersionName]

## What is the effect on users?

Requires skin support.  User can see the specific version of movie library item playing (if a version name has been set)
## Screenshots (if appropriate):
screenshot00016

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- X ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -3925,6 +3925,13 @@ const infomap musicplayer[] =    {{ "title",            MUSICPLAYER_TITLE },
 ///     @skinning_v20 **[New Infolabel]** \link VideoPlayer_HdrType `VideoPlayer.HdrType`\endlink
 ///     <p>
 ///   }
+///   \table_row3{   <b>`VideoPlayer.VideoVersionName`</b>,
+///                  \anchor VideoPlayer_VideoVersionName
+///                  _string_,
+///     @return String containing the name of the version of a video - empty if no version available
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp:3931]: Maybe make a bit more clear that this is returning the version for the currently playing video, if any.

Like: "String containing the name of the version of the currently playing video - empty if no version available."

Comment (scott967) [reply xbmc/GUIInfoManager.cpp]: OK did this then squashed/force pushed.  Maybe I should have done a --fixup commit instead?

**Comment by CrystalP** [comment]
With the current implementation of versions, you're most likely going to get the extra name in that info label when playing an extra (you should be able to confirm easily).
So depending on intent, you should exclude that case by code from the info label or name the label in a more generic way.

**Comment by ksooo** [comment]
Same applies for existing listitem.videoversioname. Would it make sense to fix implementation for both labels to actually only return the version name, not the extra name? And to introduce two new labels for extras?

**Comment by scott967** [comment]
Yes, looking at extras, that does throw a spanner into things using the title as versionname.  @ksooo I think you are right, or at least if there is a videoplayer.isvideoextra/videoplayer.hasvideoversions it might help.  As it is, in skin I have to visibility test for string.isequal(videoplayer.viderversionname,$LOCALIZE[40400]) since I guess every movie is at least a "standard edition".

**Comment by CrystalP** [comment]
> I guess every movie is at least a "standard edition".
By default yes, but nothing prevents a user from changing it to match the editions he actually has (only director's cut for example)

**Comment by CrystalP** [comment]
The same questions apply to the list items yes. For v21 it seems easier to restrict the label to return only version name.
Would a label with extra name have any use? For extras, the "title" likely contains the extra name, and it's the parent movie name that won't be found in any label. But zero skinning skills here to confirm that thought.

**Comment by scott967** [comment]
From skinning perspective see some users as not interested in versions, so a skin setting could probably suffice to make display optional, but I do see need for some users to see the version type in the UI for videoplaylist listitems and playing video.  Not so concerned about extras.

**Comment by CrystalP** [comment]
You could show the version name only when it's not the default 40400 OR the movie has multiple versions. Then only users who care should see it.

**Comment by jjd-uk** [comment]
It would be helpful is we had a `VideoPlayer.IsVersion` infobool is to the `VideoPlayer.IsStereoscopic` for 3D and for completeness a `ListItem.IsVersion`

**Comment by ksooo** [comment]
What is the meaning of “isversion”? Regarding terminology i still think that every movie *has* a version, not that certain movies *are * versions. For what kind of movie “isversion” would return true?

**Comment by jjd-uk** [comment]
There ideally needs to be a way to identify if it's one of a number of versions? As far as I'm aware this currently be done.

**Comment by MikeSiLVO** [comment]
Does every movie have to have a default version assigned?
I always thought it odd that every movie is automatically a Standard Edition.

I assume that is why there is ListItem.IsVideoExtra but no corresponding one for versions like the "Has" INFObools do?
ListItem.HasVideoVersions
ListItem.HasVideoExtras

**Comment by ksooo** [comment]
> to identify if it's one of a number of versions

Isn't that "ListItem.HasVideoVersions" ?

**Comment by CrystalP** [comment]
> > to identify if it's one of a number of versions
> 
> Isn't that "ListItem.HasVideoVersions" ?

exactly

**Comment by jjd-uk** [comment]
We still need a videoplayer identifier if we want to limit the display of VideoPlayer.VideoVersionName to only videos are one of a number of versions.

**Comment by CrystalP** [comment]
If @scott967 feels up to it, it's almost the same as adding the version name.

The questions remains whether to return a value in the version name label only for versions and blank for extras, or always return a value. I think the former makes more sense.

**Comment by ksooo** [comment]
> We still need a videoplayer identifier 

You mean "VideoPlayer.HasVideoVersions" is missing? That is correct, but can easily be changed.

**Comment by jjd-uk** [comment]
`Has` feelt wrong to me which is why I went with `Is`, but that's semantics I don't really care about, so yes `VideoPlayer.HasVideoVersions` is missing.

**Comment by jjd-uk** [comment]
> The questions remains whether to return a value in the version name label only for versions and blank for extras, or always return a value. I think the former makes more sense.

If I add this to Estuary then I would only want to display any result from `VideoPlayer.HasVideoVersions` only if it's a version, and I would not want anything for extras. So returning a blank value would suit me.

**Comment by scott967** [comment]
Not exactly sure where we are; agree that a VideoPlayer.HasVideoVersions bool would allow (with possible skin-setting options) to only display the version name when there are multiple versions in the library (was my intent).  Not sure what, if anything should display for an extra.  I think options are the "parent" or nothing.

**Comment by da-anda** [comment]
don't forget that we also have `VIDEOPLAYER_NEXT_TITLE` - so we will need the same additional infolabels and infobools for the next fileItem in queue

**Comment by CrystalP** [comment]
@scott967 sure if the extra's parent name is available, but maybe it's not (likely that "parent" name ie the item's name contains the extra name). Not a big deal I think.
testing required as things don't always look the same from the skin and in core.

**Comment by CrystalP** [comment]
For the skin change @scott967 had in mind we need VideoPlayer.HasVideoVersions and VideoPlayer.VideoVersionName

with updated logic of ListItem/VideoPlayer.VideoVersionName to return blank when not a version

it may be nice to take advantage of it in Estuary as well if it fits, I don't know what you think @jjd-uk

**Comment by da-anda** [comment]
as mentioned, some skins also show the upcoming movie/video as part of the video OSD, and if that item is a version, `VideoPlayer.Next*` labels probably will also be needed (version/extras related info bools as well as the version name)

So probably:
- videoplayer.nextIsVideoExtra
- videoplayer.nextVideoVersionName
- (videoplayer.nextHasVideoVersions - not sure this is needed for conditions to show/hide the videoVersionName)

@Hitcher did I miss one?

**Comment by CrystalP** [comment]
@da-anda I may be searching incorrectly but VIDEOPLAYER_NEXT_TITLE and friends seem to be populated only by PVR, which doesn't do versions

**Comment by Hitcher** [comment]
> @Hitcher did I miss one?

```
VideoPlayer.NextTitle
VideoPlayer.NextGenre
VideoPlayer.NextPlot
VideoPlayer.NextPlotOutline
VideoPlayer.NextStartTime
VideoPlayer.NextEndTime
VideoPlayer.NextDuration
```

That's all of the `VideoPlayer.Next` labels.

There are no bools though.

**Comment by CrystalP** [comment]
Replaced by 24695.

**Pull request closed without merging**